src/Core/Program.cs

```diff
@@ -686,7 +686,14 @@ public UserGlobal ModifyUserGlobal(IProcessorArchitecture arch, Address address,
                 this.ImageMap.AddItemWithSize(address, item);
             else
                 this.ImageMap.AddItem(address, item);
-
+            //$BUGBUG: what about x86 segmented binaries?
```

Comment:
If `address` has a selector, then you can look for the segment in `SegmentMap`.

Reply:
I tried to add it to `ImageSegment.Fields` but it is not shown at output file. Is there support for user-defined segmented global variables?

Reply:
We should add unit tests to confirm this, both at the assembler output stage and during type inference.

Reply:
Should I attach such tests to this PR?

Reply:
I can try to create test at `https://github.com/uxmal/reko/blob/7efb910c1420c51de9d0364464cd87601a8a8554/src/UnitTests/Core/Output/SegmentFilePolicyTests.cs`.

Reply:
I have added test (failed) for `SegmentFilePolicy` at 78f9de3

Reply:
Finally, I have found solution. The fix is coming.

Reply:
Done in 162c14b and 8bc0331